hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/schema/TableResolverImpl.java

```diff
@@ -159,7 +151,8 @@ private Mapping resolveMapping(Mapping mapping) {
         return new Mapping(
                 mapping.name(),
                 mapping.externalName(),
-                mapping.dataConnection(), type,
+                mapping.dataConnection(),
```

Comment:
what will happen when someone changes DATA CONNECTION type, eg. from JDBC to Kafka? Will we be able to detect it in a meaningful way?

Reply:
TDD definition is pretty clear for this and similar cases :

> When a query which uses a mapping based on a data connection is submitted, the current
settings of the data connection are used, and subsequent changes aren’t reflected, nor
do they change the query failure.

Reply:
are you sure that it is related to stored connections metadata or connection/mapping currently being used? AFAIR the intention was that when a _running_ processor uses a mapping/data connection and this object is altered, such processor continues to use version with which it had started. But there was no intention to make `data connection` only a template for creating mappings which afterwards are independent. For example, you can change password in `data connection` and all subsequently started jobs should use new password.

Reply:
> are you sure that it is related to stored connections metadata or connection/mapping currently being used?

From the wording used in TDD, I can make an assumption that it is related both to metadata and currently running jobs.

Reply:
> For example, you can change the password in the data connection and all subsequently started jobs should use new password.

Yes, and they will. I don't see any contradictions for this case - new job will use new data connection with updated password

Reply:
password - yes, but connectionType - no.

After this PR I can immediately file the following issue:
> I changed data connection (DROP+CREATE) from MySQL to Postgres, but my mappings created before the change still use MySQL dialect.

Reply:
The wording in the TDD is unclear, I clarified this with Viliam what it was supposed to mean and we agreed that it is that running queries are not affected, but new queries from the mapping take the new settings. The password update is good example, another would be host update in jdbc url.

If you use the data connection in a job, the new settings is used even when a job is restarted.